provider/oci/storage_volumes.go

```diff
@@ -527,20 +541,41 @@ func (v *volumeSource) AttachVolumes(ctx envcontext.ProviderCallContext, params
 }
 
 func (v *volumeSource) DetachVolumes(ctx envcontext.ProviderCallContext, params []storage.VolumeAttachmentParams) ([]error, error) {
+	var err error
 	ret := make([]error, len(params))
+	authProblems := false
 	instanceAttachmentMap := map[instance.Id][]ociCore.IScsiVolumeAttachment{}
+
+	var isAuthFailure = func(err error) bool {
+		return allProvidersCommon.MaybeHandleCredentialError(common.IsAuthorisationFailure, err, ctx)
+	}
+
 	for idx, param := range params {
+		// If we have already encountered a credential error, skip subsequent requests
+		if authProblems {
+			ret[idx] = errors.Trace(err)
```

Comment:
There are so many "err" instances that it's a bug waiting to happen - let's declare a "credErr" and assign any credential error to that and use it to avoid confusion.